yjit/src/core.rs

```diff
@@ -1390,6 +1390,14 @@ pub fn gen_entry_point(iseq: IseqPtr, insn_idx: u32, ec: EcPtr) -> Option<CodePt
     }
     */
 
+    // Compute the current instruction index based on the current PC
+    let insn_idx: u32 = unsafe {
+        let pc_zero = rb_iseq_pc_at_idx(iseq, 0);
+        let ec_pc = get_cfp_pc(get_ec_cfp(ec));
+        let ptr_diff = (ec_pc as usize) - (pc_zero as usize);
+        (ptr_diff / SIZEOF_VALUE) as u32
```

Comment:
```suggestion
        ec_pc.offset_from(pc_zero).try_into().ok()?
```

[ptr::offset_from](https://doc.rust-lang.org/std/primitive.pointer.html#method.offset_from) does this calculation, and we can check for overflow here with `?`.